packages/@cosmos-building-blocks/pipeline/src/source/github-enterprise-webhook.ts

```diff
@@ -13,51 +22,52 @@ export class WebhookCustomResource extends Construct {
 
     const { project, webhookFilters } = props;
 
-    const projectName = Lazy.stringValue({
+    const role = project.role as Role;
+
+    const projectName = Lazy.string({
       produce: () => project.projectName,
     });
-    const projectArn = Lazy.stringValue({
+    const projectArn = Lazy.string({
       produce: () => project.projectArn,
     });
 
-    const defaultAwsCall = {
-      service: 'CodeBuild',
-      physicalResourceId: PhysicalResourceId.of(`${projectName}Webhook`),
-      parameters: {
-        projectName: projectName,
-        buildType: 'BUILD',
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        filterGroups: webhookFilters.map((x: any) => x._toJson()),
-      },
-    };
+    const serviceProvider = CustomResourceProvider.getOrCreateProvider(this, 'Custom::WebhookCustomResource', {
+      codeDirectory: `${__dirname}/github-enterprise-webhook-handler`,
+      runtime: CustomResourceProviderRuntime.NODEJS_12,
+      role: role,
```

Comment:
Might wanna check why the role was not provided as a prop in cdk...may be to handle a specific scenario where one build might end up using same role created by another build